### ISSUE TITLE
Check that remove inverse ops (reshapes/transposes/etc.) works for conv2d 

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-forge-fe#133]
Dependent on this issue first:
- https://github.com/tenstorrent/tt-forge-fe/issues/199

**Comment by LPanosTT** [comment]
I've begun working on a document that describes this process and how we might want to do it in MLIR: https://docs.google.com/document/d/1zYmTzFqeMHydJz5_z3WYZ7suFSfqya7NO7Z7WNRqek0/edit?usp=sharing

**Comment by LPanosTT** [comment]
@nvukobratTT I think this issue should probably belong to the MLIR repo since this logic will be written there when I get around to it.

**Comment by nvukobratTT** [comment]
> @nvukobratTT I think this issue should probably belong to the MLIR repo since this logic will be written there when I get around.

For now, I'd like to keep it here as well. 

We want to keep support for this pass until it's fully translated into MLIR. Once we start with MLIR planning, we can also open new issues. But I suppose that will be part of followup milestones :))

Let me know your thoughts

**Comment by LPanosTT** [comment]
In that case, this logic already exists in forge, save some minor changes I've done in PyBuda in the last couple months. Also, we are not decomposing conv/pool using reshapes in forge anymore, that part is now done in MLIR. Which means the erase inverse ops logic must also go there.

**Comment by nvukobratTT** [comment]
> In that case, this logic already exists in forge, save some minor changes I've done in PyBuda in the last couple months.
Yup, that's fine, my point of this issue is just to make sure that this pass continues to run. We had a most benefits from it from conv2d decomposed. However, there are still benefits of having it for LLM models as well. 

> Also, we are not decomposing conv/pool using reshapes in forge anymore, that part is now done (or will be done) in MLIR. Which means the erase inverse ops logic must also go there.
And I agree, now that conv2d is now in lower parts, we should for sure move this pass to MLIR sooner or latter.
